tests/utils.py

```diff
@@ -84,14 +86,15 @@ def read_dependencies(distribution: str) -> PackageDependencies:
     with Path("stubs", distribution, "METADATA.toml").open("rb") as f:
         dependencies = tomli.load(f).get("requires", [])
     assert isinstance(dependencies, list)
-    typeshed, external = [], []
+    typeshed: set[str] = set()
+    external: set[str] = set()
     for dependency in dependencies:
         assert isinstance(dependency, str)
         maybe_typeshed_dependency = Requirement(dependency).name
         if maybe_typeshed_dependency in pypi_name_to_typeshed_name_mapping:
-            typeshed.append(pypi_name_to_typeshed_name_mapping[maybe_typeshed_dependency])
+            typeshed.add(pypi_name_to_typeshed_name_mapping[maybe_typeshed_dependency])
         else:
-            external.append(dependency)
+            external.add(dependency)
```

Comment:
I'm not sure deduplication here is needed. We'll only have duplicate packages here if a package has something like `requires = ["numpy", "numpy"]` in its `METADATA.toml` file. And that seems pretty unlikely, no?

Reply:
Right, and `requires = ["numpy", "numpy"]` is exactly how I checked for duplication ^^" Makes sense.

I'll move the de-duplication to the right spot.